torch/csrc/jit/frontend/ir_emitter.cpp

```diff
@@ -2896,6 +2896,39 @@ struct to_ir {
         }
         return iterable_tree;
       }
+      case prim::list: {
+        if (apply.inputs().size() == 0) {
+          TypePtr type = type_hint ? type_hint : ListType::ofTensors();
```

Comment:
```
ListTypePtr ListType::ofTensors() {
  static auto value = ListType::create(TensorType::get());
  return value;
}
```

Maybe `ListType::create(TensorType::getInferred())` should be used (or a function that returns this) in this case since the type is being inferred.

Reply:
Agreed this would be nice, but imo it's a little awkward to have List[Inferred Tensor]. We're inferring the list itself. We do this other places like `x = []` or `x = {}`, and would be nice to add as a follow.